pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -198,6 +200,22 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 		}
 	}
 
+	// Check if Status is empty and status annotation is present
+	// Manually restore data.
+	if r.hostHasStatus(host) {
+		reqLogger.Info("Fetching Status from Annotation")
+		objStatus := r.getHostStatusFromAnnotation(host)
+		if objStatus != nil {
+			host.Status = *objStatus
+			errStatus := r.client.Status().Update(context.TODO(), host)
+			if errStatus == nil {
+				return reconcile.Result{Requeue: true, RequeueAfter: 0}, nil
```

Comment:
We don't need to explicitly requeue here because the status update will trigger that.

Reply:
Done